homework/list.js

```diff
@@ -1,79 +1,90 @@
+      ul.id = 'note-title-list';
+      var buff = document.createDocumentFragment();
+      list.forEach(function(note, i) {
+        var li = document.createElement('li');
+        li.dataset.noteId = i;
+        li.classList.add('note-title');
+        li.textContent = note.title;
+        // Note: buff is captured, so we now have a
+        // little closure naturally.
+        buff.appendChild(li);
+      });
+      ul.appendChild(buff);
+      this._wrapper.appendChild(ul);
+    },
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
 
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
+    onNoteOpen(event) {
+      if (event.target.classList.contains('note-title')) {
+        var id = event.target.dataset.noteId;
+        var content = this._listNoteContent[id];
+        window.dispatchEvent(new CustomEvent('note-open',
+          { detail: content }));
+      };
+    },
 
-  function preloadFirstNote() {
-    if (_listNoteContent.length !== 0) {
-      var content = _listNoteContent[0];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    }
-  }
+    preloadFirstNote() {
+      if (this._listNoteContent.length !== 0) {
+        var content = this._listNoteContent[0];
+        window.dispatchEvent(new CustomEvent('note-open',
+          { detail: content }));
+      }
+    },
 
-  function updateList(list) {
-    _listNoteContent = list;
-  }
+    updateList(list) {
+      this._listNoteContent = list;
+    },
 
-  function drawList() {
-    var list = _listNoteContent;
-    var ul = document.createElement('ul');
-    ul.id = 'note-title-list';
-    var buff = document.createDocumentFragment();
-    list.forEach(function(note, i) {
-      var li = document.createElement('li');
-      li.dataset.noteId = i;
-      li.classList.add('note-title');
-      li.textContent = note.title;
-      // Note: buff is captured, so we now have a
-      // little closure naturally.
-      buff.appendChild(li);
-    });
-    ul.appendChild(buff);
-    _wrapper.appendChild(ul);
-  }
+    fetchList(afterFetch) {
+      return new Promise((function(resolve,reject){
+      var xhr = new XMLHttpRequest();
+      xhr.open('GET', 'http://127.0.0.1:8000/demo-list-notes.json', true);
+      xhr.responseType = 'json';
+      xhr.onreadystatechange = function(e) {
+        // Watch out: we have a mysterious unknown 'this'.
+        if (this.readyState === 4 && this.status === 200) {
+          var listData = this.response;
+          // The flow ends here.
+          afterFetch(listData);
```

Comment:
**WRONG**: You either keep it as a callback-style function (so you **won't** get the point) OR re-write it as a Promise-based function (that's what I need you to complete). But to wrap it inside a Promise **without resolving or rejecting the result** is showing you didn't understand Promise after all. Please, check the MDN link here:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise#Examples

Or if you like, you can look up any other examples on the Internet.